### PR TITLE
Fix warning on Pickles

### PR DESCRIPTION
**Pull request by rbonichon**
This cleans up the newly introduced warnings by the PRs fixing #13482

### CODE REVIEW

**Comment by dannywillems** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
The CI job `dev unit-tests` seems to confirm the warnings are not there anymore.

File: src/lib/pickles/wrap.mli [hunk 1]
```diff
@@ -1,162 +1,132 @@
...
-             Snarky_backendless.Snark_intf.Boolean0.t )
-           Pickles_types.Plonk_types.Opt.t
-         , ( Impls.Wrap.Impl.Field.t Composition_types.Scalar_challenge.t
-             Composition_types.Wrap.Proof_state.Deferred_values.Plonk.In_circuit
-             .Lookup
-             .t
-           , Impls.Wrap.Impl.field Snarky_backendless.Cvar.t
-             Snarky_backendless.Snark_intf.Boolean0.t )
-           Pickles_types.Plonk_types.Opt.t
+       Kimchi_pasta_basic.Pallas.Affine.t Plonk_verification_key_evals.t
```
Comment (dannywillems) [inline src/lib/pickles/wrap.mli:13]: I would suppose we want to keep the curve name "abstract" with Tick/Tock.

File: src/lib/pickles/wrap.mli [hunk 2]
```diff
@@ -1,162 +1,132 @@
...
+           , (int64, Nat.N4.n) Vector.vec
            , ( 'b
-             , ( Kimchi_pasta.Pallas_based_plonk.Proof.G.Affine.Stable.V1.t
+             , ( Backend.Tock.Proof.G.Affine.t
                , 'actual_proofs_verified )
-               Pickles_types.Vector.t
-             , ( ( Import.Challenge.Constant.t Import.Scalar_challenge.t
-                   Import.Bulletproof_challenge.t
+               Vector.vec
+             , ( ( Challenge.Constant.t Kimchi_types.scalar_challenge
```
Comment (dannywillems) [inline src/lib/pickles/wrap.mli:53]: Why this change?

**Comment by dannywillems** [comment]
Connection to [#13724](https://github.com/MinaProtocol/mina/issues/13724)

**Comment by rbonichon** [comment]
!ci-build-me

File: src/lib/pickles/wrap.mli (hunk 1 above)
Comment (rbonichon) [reply src/lib/pickles/wrap.mli]: I propose to address this in https://github.com/MinaProtocol/mina/pull/14081

File: src/lib/pickles/wrap.mli (hunk 2 above)
Comment (rbonichon) [reply src/lib/pickles/wrap.mli]: Same, can we do that in https://github.com/MinaProtocol/mina/pull/14081?

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
There are still some warnings:
```


Warning: Consider passing '-g' option to ocamlc.
--


<span class="term-fg35 term-fg1">Warning</span> 27 [unused-var-strict]: unused variable poly_fp.
```

Is it expected?

**Comment by rbonichon** [comment]
The remaining warnings (and the re-addition of the warnings to dune files) are handled in #14080. 
The reason is the change in type definitions impacts a CI task tracking versioned types, which will fail. #14080 is thus to check that this specific failure is not an actual issue, whereas the warnings resolved in this PR have no such issue.

**Review by dannywillems — APPROVED** [review]

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me
